integration-tests/itest/deploy_test.go

```diff
@@ -0,0 +1,306 @@
+package main_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strings"
+	"time"
+
+	"os/exec"
+
+	"lightbend.com/cloudflow/itest/cli"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// Assumptions:
+// In this test, we are going to deploy the test application to an existing cluster.
+// For this first step, we assume that the cluster is correctly deployed and that kubectl is configured to talk to it.
+// The application is already built
+// We also assume that the test is going to be executed manually.
+// Automation will be the next step
+
+const (
+	JsonTokenSrc    = "/keybase/team/assassins/gcloud/pipelines-serviceaccount-key-container-registry-read-write.json"
```

Comment:
What ere these credentials required for?
Is there a way we can decouple this so people outside of the core Cloudflow team, including the CI server, can also run them? Can we inject credentials through an environment var?

Reply:
This is the JSON Token for the Docker repo. I think that we should externalize this in the scope of "running test on CI" - This should work for the team in the current "run manual" mode.

Reply:
So you are saying this is a first attempt and you agree that we need to externalize this in subsequent steps, right?
I can agree with that but what I want to prevent is us writing dozens of these tests with hardcoded, team-only credentials. So I suggest we pick up this particular issue as part of the next step, which is "Ci-ification" of this test

I would also urge us to use public repos instead of protected repos t sidestep the entire issue, unless we explicitly want to test registry authentication aspects of the CLI.